test/parallel/test-assert.js

```diff
@@ -58,6 +58,28 @@ a.equal(null, undefined);
 a.equal(true, true);
 a.equal(2, '2');
 a.notEqual(true, false);
+assert.throws(() => {
+  a.equal(0, []);
+}, {
+  actual: 0,
+  expected: [],
+  operator: 'equal',
+  generatedMessage: true
+});
+assert.throws(() => {
+  a.equal({ [Symbol.toPrimitive]() { return 5; } }, 5);
```

Comment:
I guess this is the edge case that this is for? Is there a simpler/more common one? (It's OK if there's not. I'm just making sure I understand this change!)

Reply:
The above one is the more common one: `0 == []`. This is IMHO quite bad. I can also add e.g., `[1] == true` or `['0'] == 0` (`true`, `1`, `'1'`, `1n`, etc. are all interchangeable, just as `false`, `0`, `''`, `'0'`...).